src/deploy/functions/backend.ts

```diff
@@ -0,0 +1,356 @@
+import * as proto from "../../gcp/proto";
+import * as gcf from "../../gcp/cloudfunctions";
+import * as cloudscheduler from "../../gcp/cloudscheduler";
+import * as utils from "../../utils";
+import { FirebaseError } from "../../error";
+import { Context } from "./args";
+
+export interface ScheduleRetryConfig {
+  retryCount?: number;
+  maxRetryDuration?: proto.Duration;
+  minBackoffDuration?: proto.Duration;
+  maxBackoffDuration?: proto.Duration;
+  maxDoublings?: number;
+}
+
+export interface PubSubSpec {
+  id: string;
+  project: string;
+
+  // What we're actually planning to invoke with this topic
+  targetService: FunctionSpec;
```

Comment:
It strikes me as odd to embed the entire `FunctionSpec` here instead of just an identifier or reference.